pyresample/area_config.py

```diff
@@ -849,3 +850,37 @@ def convert_def_to_yaml(def_area_file, yaml_area_file):
     with open(yaml_area_file, 'w') as yaml_file:
         for area in areas:
             yaml_file.write(area.create_areas_def())
+
+
+def generate_area_def_rst_list(area_file):
+    """Create rst list of available area definitions with overview plot.
+
+    Args:
+        area_file (str): Path to area yaml file.
+
+    Returns:
+        str: rst list formatted string.
+    """
+    area_list = []
+
+    template = ("{area_name}\n"
+                "{n:^>{header_title_length}}\n\n"
+                ".. raw:: html\n\n"
+                "{content}\n\n"
+                "     <hr>\n\n")
+
+    for i, (aname, params) in enumerate(_read_yaml_area_file_content(area_file).items()):
+        area = _create_area_def_from_dict(aname, params)
+        if hasattr(area, "_repr_html_"):
+            if i == 0:
+                area_rep = area_repr(area, include_header=False, include_static_files=False)
+            else:
+                area_rep = area_repr(area, include_header=False)
+
+            content = "\n".join([x.rjust(len(x) + 5) for x in area_rep.split("\n")])
+            area_list.append(template.format(area_name=aname, n="", header_title_length=len(aname),
+                                             content=content))
+        else:
+            pass
```

Comment:
How about swapping the `if` logic and doing `if not `hasattr(area, "_repr_html_"): continue` and then dedenting the main area repr portion of this?

Reply:
Yes, very nice I will change that :-).